### PR TITLE
#P09 As a participant - I would like to submit my entry. 

### PR DESCRIPTION
**Pull request by ClaireLally8**
## Description

Creates the app for the submissions of completed hackathon project.  Utilises the HackProject model and is filled out with a form by the users. 
- Created a forms.py file within the submissions app which then is used by the crispy-forms template to fill the model. 
- Two templates - submit.html and success.html
- A view that checks if the request is a GET or POST.    If it's a GET request, it will render the submit.html page,  if its a post, it will take the data from the form, checks if it's valid, saves it into the model and then renders the success.html page. 

## Pull request type

- [ ] #M01 (Miscellaneous User Story #01)
- [ x] #P02 (Participant User Story #02) - #PO9 As a participant, I would like to submit my hackathon entry. 
## Testing
Manual testing by checking it functions.  
Automated testing - * crickets * 

## Additional Information

Will need the templates edited as these are not finalised.  Found within the submissions/templates/submissions folder :)

## Does this introduce a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: hackathon/models.py
Comment (Eventyret) [inline hackathon/models.py:113]: This would be missing the triple quotes at the start to make it a docstring.

File: submissions/views.py
Comment (Eventyret) [inline submissions/views.py:5]: Could we please add the missing docstring for this function :)

File: main/settings.py
Comment (TravelTimN) [inline main/settings.py:32]: Somehow missed this - can these comments be removed on the various lines? Not needed ultimately.

File: main/settings.py
Comment (TravelTimN) [inline main/settings.py:34]: Here, too

File: submissions/forms.py
Comment (TravelTimN) [inline submissions/forms.py:1]: Do we need a blank line at the start?

File: submissions/forms.py
Comment (TravelTimN) [inline submissions/forms.py:30]: This file needs a blank line at the end, please.

File: submissions/forms.py
Comment (TravelTimN) [inline submissions/forms.py:16]: Odd combination of 'single' quotes, and "double" quotes.
Let's try to keep it consistent, the majority of the app so far has "double" I think.

Also, I think "Speaker" singular is fine, we only ask that one person speaks during the event.

File: submissions/forms.py
Comment (TravelTimN) [inline submissions/forms.py:18]: GitHub with capital H here please 🙏

File: submissions/urls.py
Comment (TravelTimN) [inline submissions/urls.py:6]: This file needs a blank line at the end.

File: submissions/views.py
Comment (TravelTimN) [inline submissions/views.py:6]: No #comment here needed at the end.

File: submissions/views.py
Comment (TravelTimN) [inline submissions/views.py:10]: We could also consider using the **messages.success** flash notification?
"Thanks! Your project has been successfully submitted!"

File: submissions/views.py
Comment (TravelTimN) [inline submissions/views.py:18]: Missing a blank line at the end of this file.

**Review by TravelTimN — CHANGES_REQUESTED** [review]
Looking great!
A few minor adjustments outlined below! 🦗

File: hackathon/models.py
Comment (stefdworschak) [inline hackathon/models.py:116]: This looks more than 79 characters. Can you split the line up please?
Also, please remove the null=True as you are already defaulting it to an empty string which I would keep.

File: hackathon/models.py
Comment (stefdworschak) [inline hackathon/models.py:141]: Same here with the line length.

File: submissions/apps.py
Comment (stefdworschak) [inline submissions/apps.py:4]: Sorry to be nitpicky, but I would keep with the naming convention and call prefix your `Submission` classes with `HackProject` to make it `HackProjectSubmissionConfig` and `HackProjectSubmissionForm`.

File: submissions/templates/submissions/success.html
Comment (stefdworschak) [inline submissions/templates/submissions/success.html:5]: Maybe flesh out the success message a little and add a link back to some other page.

File: submissions/views.py
Comment (stefdworschak) [inline submissions/views.py:14]: Here I am not sure you need to create an extra context dict with only one key, I would just pass this directly into the render function.

**Comment by stefdworschak** [comment]
@ClaireLally8 sorry, I did not want to pile on and be the 3rd person to review your pull request, but since we were talking about your PR a bit I would be amiss not to review it for you!! Also, only for a few minor things I think it looks really good! 🥇

File: submissions/forms.py
Comment (ClaireLally8) [reply submissions/forms.py:16]: I call it "inconsistency'

**Comment by ClaireLally8** [comment]
Thanks y'all @stefdworschak @TravelTimN @Eventyret I guess I'll fix these at some stage today ;)   (but please, be perfectionists,  I need to get better at the refining stuff!)

**Comment by ClaireLally8** [comment]
If you don't approve this  *shakes fist *

File: hackathon/models.py
Comment (stefdworschak) [inline hackathon/models.py:2]: Can you remove this line seeing as we are using the CustomUser below?

File: hackathon/models.py
Comment (stefdworschak) [inline hackathon/models.py:4]: As per PEP8 could you please add a blank line between 3rd party and local modules?

File: hackathon/models.py
Comment (stefdworschak) [inline hackathon/models.py:123]: I think you can remove the `default=""` here, this is only needed for Char or TextField.

File: hackathon/models.py
Comment (stefdworschak) [inline hackathon/models.py:130]: Do we need this here? Or more specifically can the person submitting the project legally agree to the sharing permissions for all team members? I would say we put something in the T&Cs when signing up that they need to confirm that any work that is produced as part of participating on this platform can be shared. @TravelTimN what do you think about this?

File: hackathon/models.py
Comment (stefdworschak) [inline hackathon/models.py:152]: Again, you can remove the `default=""`

**Review by stefdworschak — APPROVED** [review]
@ClaireLally8 gonna approve this now since you are threatening me (yes, I am referring to the fist shaking lol). Please have a look at my final few comments and I will merge it when you revert back.

File: hackathon/models.py
Comment (ClaireLally8) [reply hackathon/models.py:130]: Yeah @TravelTimN

**Comment by stefdworschak** [comment]
@ClaireLally8 @TravelTimN I am going to merge this now and will create a follow up issue for ourselves to decide this.

**Pull request merged**
